### PR TITLE
[#267] do not add duplicated comments asking for approval

### PR DESCRIPTION
**Pull request by psiroky**
This seems to fix the issue (tested locally with simple setup). That being said, I am not sure this is the best approach. Any comments are more than welcome.

edit by @samrocketman:
See also #267

### CODE REVIEW

**Comment by ge0ffrey** [comment]
@benpatterson this PR fixes jenkins spam for us. If we rebase to get the merge conflicts fixed introduced recently, any chance it can be merged soonish?

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 1]
```diff
@@ -129,20 +129,46 @@ public GhprbPullRequest(GHPullRequest pr,
...
+    private boolean containsComment(GHPullRequest ghPullRequest, String expectedBody) {
+        List<GHIssueComment> prComments = null;
+        try {
+            prComments = ghPullRequest.getComments();
+        } catch (IOException e) {
+            logger.log(Level.WARNING, "Failed to get comments for PR " + ghPullRequest, e);
+            return false;
+        }
+        for (GHIssueComment comment : prComments) {
+            if (comment.getBody() != null && comment.getBody().contains(expectedBody)) {
```
Comment (bjoernhaeuser) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: i think contains is not the thing we are looking for. the body should exactly match the expected content, no?

Comment (psiroky) [reply src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: Good point, I will fix that.

**Comment by psiroky** [comment]
Yes, checking also an author of the comment would be ideal. Not sure if that info is available at that point, but will check.

**Comment by psiroky** [comment]
@bjoernhaeuser I've changed the `contains()` to `equals()`. I haven't found a way to check the author of the comment as that value does not seems to be reliably accessible (e.g. when you use the auth. token, it's just the token; without username).

**Comment by bjoernhaeuser** [comment]
I will try to go through this over the weekend.

**Comment by psiroky** [comment]
Hello, any update on this? I can rebase the PR again, but it only makes sense if there is real promise that it will processed :)

**Comment by samrocketman** [comment]
Hi @psiroky, if you rebase/update based on the latest master I can take a crack at reviewing, testing, and merging if all looks good.

**Comment by psiroky** [comment]
@samrocketman rebased! Thanks for looking at this.

**Pull request merged**